### DISCUSSION TITLE
Field filter does not work as expected

### DISCUSSION THREAD

**Discussion by zim32** [tokio-rs/tracing#2588] [Q&A] [answered]
## Bug Report
When I try to filter spans by field, nothing is shown.

F.e.: 
`RUST_LOG=[loop_step]=debug cargo run`
This shows logs like this:
INFO loop_step: raft::node: Follower in term 1 cat="node"

This also works:
`RUST_LOG=[{cat}]=debug cargo run`
INFO loop_step: raft::node: Follower in term 1 cat="node"

But this does not show anything:
`RUST_LOG=[loop_step{cat}]=debug cargo run`

This does not work too:
`RUST_LOG=[loop_step{cat=\"node\"}]=debug cargo run`

I expect to see the same line  as above filters:
INFO loop_step: raft::node: Follower in term 1 cat="node"

### Version

```
│   └── tracing v0.1.37
│       ├── tracing-attributes v0.1.23 (proc-macro)
│       └── tracing-core v0.1.30
│   │   │   │   └── tracing v0.1.37 (*)
│   │   │   ├── tracing v0.1.37 (*)
│   │   │   └── tracing v0.1.37 (*)
│   └── tracing v0.1.37 (*)
├── tracing v0.1.37 (*)
├── tracing-subscriber v0.3.17
│   ├── tracing v0.1.37 (*)
│   ├── tracing-core v0.1.30 (*)
│   └── tracing-log v0.1.3
│       └── tracing-core v0.1.30 (*)
```

### Platform

x86_64 GNU/Linux
Instead, this happened: <explanation>
-->

### Details

My usage pseudocode is:

`tokio::spawn(async move {...}.instrument(span!(Level::INFO, "loop_step")))`

**Comment by zim32** [comment, marked as answer]
Ah.. It seems like usage are a little bit confusing.

[loop_step{cat}] tryies to filter SPANS with field cat, not EVENTS.

**Reply by hawkw** [reply]
Yeah, that's correct. Field value filters filter whether events inside of a span are enabled or not, based on the span's field values. We've tried to describe this in the documentation, but if you have any suggestions for making it clearer, please let us know!
